eclass/gap-pkg.eclass

```diff
@@ -0,0 +1,408 @@
+# to specify it yourself. Compiled packages will likely need
+# sci-mathematics/gap in DEPEND as well, and may also want a subslot
+# dependency.
+
+case ${EAPI} in
+	8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
+esac
+
+# Some packages have additional homepages, but pretty much every GAP
+# package can be found at this URL.
+HOMEPAGE="https://www.gap-system.org/Packages/${PN}.html"
+
+# _GAP_PKG_IS_REQUIRED is an internal variable that indicates whether or
+# not $PN is one of the four "required" GAP packages that are always
+# loaded, even when GAP is started with the "-A" flag. We treat this
+# four somewhat differently since they are implicit dependencies of
+# everything else in the GAP ecosystem.
+_GAP_PKG_IS_REQUIRED=no
+case "${PN}" in
+	 gapdoc|smallgrp|primgrp|transgrp)
+		 _GAP_PKG_IS_REQUIRED=yes
+		 ;;
+	 *)
+		;;
+esac
+
+# _GAP_PKG_RDEPEND is an internal variable to hold the RDEPEND entries
+# added by this eclass. We use a separate variable for this because we
+# need its contents later in gap-pkg_enable_tests, and that function is
+# called from an ebuild context where the list of RDEPEND is maintained
+# separately. Basically: the values we add to RDEPEND here do not appear
+# in RDEPEND when gap-pkg_enable_tests is called.
+_GAP_PKG_RDEPEND="sci-mathematics/gap"
+
+# The four "required" packages depend only on GAP itself, while every
+# other package depends (also) on the four required ones.
+if [[ "${_GAP_PKG_IS_REQUIRED}" = "no" ]]; then
+		_GAP_PKG_RDEPEND+="
+			dev-gap/gapdoc
+			dev-gap/smallgrp
+			dev-gap/primgrp
+			dev-gap/transgrp"
+fi
+RDEPEND="${_GAP_PKG_RDEPEND}"
+
+# @FUNCTION: gap-pkg_dir
+# @USAGE:
+# @DESCRIPTION:
+# The directory into which the gap package should be installed. The
+# accepted current location is /usr/$(get_libdir)/gap/pkg, but
+# technically this depends on the econf call in sci-mathematics/gap.
+gap-pkg_dir() {
+	echo "/usr/$(get_libdir)/gap/pkg/${PN}"
+}
+
+# @FUNCTION: _gap-pkg_gaproot
+# @INTERNAL
+# @USAGE:
+# @DESCRIPTION:
+# The directory containing sysinfo.gap. This is frequently passed to GAP
+# packages via ./configure --with-gaproot or as a positional argument to
+# the old-style fake configure scripts. We also use it to find the value
+# of $GAParch, which is contained in sysinfo.gap. The "gaproot" is
+# implicitly determined by the econf call in sci-mathematics/gap. As a
+# result, calling this function requires sci-mathematics/gap at
+# build-time.
+_gap-pkg_gaproot() {
+	echo "${ESYSROOT}/usr/$(get_libdir)/gap"
+}
+
+# @FUNCTION: _gap-pkg_gaparch
+# @INTERNAL
+# @USAGE:
+# @DESCRIPTION:
+# Return the variable $GAParch from sysinfo.gap. GAP packages put their
+# executables and libraries in bin/$GAParch, and to accomplish that, we
+# sometimes need to pass $GAParch to the build system.
+_gap-pkg_gaparch() {
+	# Use a subshell so we don't pollute the environment.
+	( . $(_gap-pkg_gaproot)/sysinfo.gap && echo "${GAParch}" )
```

Comment:
I'm glad you caught this, because while trying to test it, it became clear that nobody is using this function after https://github.com/gentoo/gentoo/pull/34472/commits/eb42fd74911e3bf2959183cbc51b9e77cdf33a74. I dropped it entirely.